### ISSUE TITLE
Delete from a variable in batch mode

### ISSUE THREAD

**Issue by MarcelPitch** [orientechnologies/orientdb#3110]
Hi all,

On 2.0-M3, I try to delete edges in a batch from a sub-query stored in variable, as follow :

```sql
BEGIN
LET groupVertices = SELECT UNIONALL(@this) AS groups FROM (SELECT FROM StandardGroup WHERE first(in('HAS_GROUP')).uid = '5ed320a8-af88-4b0a-98f9-e64c3f143ed9')
LET removeRoleEdge = DELETE edge HAS_SUBMIT_ROLE WHERE out IN first($groupVertices).groups AND in.@class = 'Thread'
COMMIT 
RETURN $removeRoleEdge
```

This command does not delete any edge.

I tried the same query, replacing `$groupVertices` with the sub-query :
```sql
first(SELECT UNIONALL(@this) AS groups FROM (SELECT FROM StandardGroup WHERE first(in('HAS_GROUP')).uid = '5ed320a8-af88-4b0a-98f9-e64c3f143ed9')).groups
```
and it works.

The problem has been reported on Google groups :

https://groups.google.com/forum/#!topic/orient-database/0uv7LaNpF5k

Have a nice week-end

**Comment by AlainBod** [comment]
I believe the following case is related to the same bug:
```
begin
let v0 = UPDATE Word SET txt="foo" UPSERT RETURN AFTER WHERE txt="foo"
CREATE EDGE Includes FROM ( SELECT FROM #12:2 WHERE $v0 NOT IN out('Includes') ) TO $v0
return null
end
```

The CREATE is supposed to create the edge only if it is not already existing. It works outside a batch but within the batch the edge is always created, as it the $v0 variable was not replaced by its value.

The following batch works:
```
begin
let v0 = UPDATE Word SET txt="foo" UPSERT RETURN AFTER WHERE txt="foo"
CREATE EDGE Includes FROM #12:2 TO $v0
return null
end
```

And the standalone command works:
```
CREATE EDGE Includes FROM ( SELECT FROM #12:2 WHERE #13:4 NOT IN out('Includes') ) TO #13:4
```
but not within a batch.

Alain

**Comment by luigidellaquila** [comment]
I fixed this in 2.0.x branch, could you please verify if my fix completely cover your use case? (it's not so easy to reproduce your schema...)

In case please reopen
